indico/modules/events/timetable/controllers/manage.py

```diff
@@ -112,3 +113,35 @@ def _process_PATCH(self):
             with track_time_changes():
                 update_timetable_entry(self.timetable_entry, updates)
         return jsonify()
+
+    def _process_DELETE(self):
+        """Delete a timetable entry"""
+        if self.timetable_entry.type == TimetableEntryType.SESSION_BLOCK:
+            delete_session_block(self.timetable_entry.session_block)
+        else:
+            delete_timetable_entry(self.timetable_entry)
+
+
+class RHTimetableBalloon(RHManageTimetableBase):
+    """Create the appropriate timetable entry balloon depending on the entry type."""
+    def _checkParams(self, params):
+        RHManageTimetableBase._checkParams(self, params)
+        self.timetable_entry = None
+        if 'timetable_entry_id' in request.view_args:
+            self.timetable_entry = (self.event_new.timetable_entries
+                                    .filter_by(id=request.view_args['timetable_entry_id'])
+                                    .first_or_404())
+        self.editable = request.args.get('editable')
```

Comment:
I usually prefer `self.editable = (request.args.get('editable') == '1')` since otherwise `editable=0` in the URL would also be considered true